src/app/pipes/DefaultNullOrEmptyPipe.ts

```diff
@@ -8,22 +8,23 @@ export class DefaultNullOrEmpty implements PipeTransform {
     switch (typeof value) {
       case 'string':
         if (value.trim().length > 0) {
-
-          if (!/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(value)) {
+          if (/\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z/.test(value) || /(\d{4})-(\d{2})-(\d{2})/.test(value)) {
```

Comment:
Can we not use angular date pipe  for this implementation. Hence no need for this code and associated test

Reply:
I did this pipe because for each empty or null field (being it string/date string/boolean, etc) we have to display hyphen. If we use date pipe we have to do the empty/null check for each field in the html template. This was an idea of optimisation I discussed with @m-ujaffer

Reply:
Irina, checking for an empty/null DATE filed and using a date pipe is still a much simpler code than producing lots of re-engineered code. I see no benefit. I agree with other string/null fields concept